### PR TITLE
Add the implementation in kernelci.data.kernelci_api.submit() to use PUT

### PR DESCRIPTION
**Pull request by JenySadadia**
Fixes: https://github.com/kernelci/kernelci-api/issues/66

kernelci/data/kernelci_api.py: Add put handler
kernelci/data/kernelci_api.py: Add logic in submit() to use PUT
in case the object already exists

Signed-off-by: Jeny Sadadia <jeny.sadadia@gmail.com>

### CODE REVIEW

File: kernelci/data/kernelci_api.py [hunk 1]
```diff
@@ -69,6 +69,12 @@ def get_node(self, node_id):
     def get_node_from_event(self, event):
         return self.get_node(event.data['id'])
 
+    def _put(self, path, data=None):
```
Comment (gctucker) [inline kernelci/data/kernelci_api.py:72]: Maybe move it next to `_get()` and `_post()` to keep the primitive HTTP wrappers together

Comment (JenySadadia) [reply kernelci/data/kernelci_api.py]: Okay.

**Review by gctucker — APPROVED** [review]
LGTM.  I'll try to use it with the related API changes to manually check it's working as expected, then it should be ready to be merged.

**Review by gctucker — APPROVED** [review]

**Comment by JenySadadia** [comment]
@gctucker We need to merge changes from this PR to 'api-pipeline' branch to fix https://github.com/kernelci/kernelci-pipeline/issues/7 issue.
The reason is we are using that branch for kernelci-pipeline.
```
# Install kernelci-core from api-pipeline topic branch
RUN pip install git+https://github.com/kernelci/kernelci-core.git@api-pipeline
```

**Comment by gctucker** [comment]
Well we need to stop using the `api-pipeline` branch now as I believe all the required changes have been merged in `main`.

**Pull request merged**